pkg/client/csw.go

```diff
@@ -0,0 +1,94 @@
+package client
+
+import (
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/pdok/pdok-metadata-tool/pkg/model/csw"
+)
+
+type CswClient struct {
+	host     *url.URL
+	endpoint string
+	client   *http.Client
+}
+
+func NewCswClient(host *url.URL) CswClient {
+	client := &http.Client{
+		Timeout: 20 * time.Second,
+	}
+	return CswClient{
+		host:   host,
+		client: client,
+	}
+}
+
+func (c CswClient) GetRecordById(uuid string, logPrefix string) (csw.MDMetadata, error) {
+	cswUrl := c.host.String() +
+		"?service=CSW" +
+		"&request=GetRecordById" +
+		"&version=2.0.2" +
+		"&outputSchema=http://www.isotc211.org/2005/gmd&elementSetName=full" +
+		"&id=" + uuid + "#MD_DataIdentification"
+
+	cswResponse := csw.GetRecordByIdResponse{}
+	err := getUnmarshalledXMLResponse(&cswResponse, cswUrl, "GET", nil, *c.client, logPrefix)
+	if err != nil {
+		return csw.MDMetadata{}, err
+	}
+	cswResponse.MDMetadata.SelfUrl = cswUrl
+	return cswResponse.MDMetadata, nil
+}
+
+// TODO Use this for harvesting service metadata in ETF-validator-go
+func (c CswClient) GetRecords(constraint *csw.GetRecordsCQLConstraint, offset int, logPrefix string) ([]csw.SummaryRecord, int, error) {
```

Comment:
Waarom hebben we `logPrefix` nodig? Dit lijkt op een arbitraire toevoeging om logging te genereren. Dit vervuild ook de logging output. Eigenlijk zou er een verbosity geconfigureerd moeten worden. Op basis daarvan kunnen we dan besluiten wel of niet te loggen. Anders zou gebruik in kangaroo heel veel logging gaan genereren.